### PR TITLE
add more test

### PR DESCRIPTION
**Pull request by bobotu**

### CODE REVIEW

File: surf/surf_test.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	checkSuRF(t, &s2, keys, vals)
+}
+
+// max key length is `initLen * (round + 1)`
+// max result size is (initSize + initSize * (round + 1)) * (round + 1) / 2
+// you can use small round (0 is allowed) to generate a sparse key set,
+// or use a large round to generate a key set which has many common prefixes.
+func randomKeys(initSize, initLen, round int) [][]byte {
+	keys := make([][]byte, initSize)
+	rand := rand.New(rand.NewSource(0xdeadbeaf))
```
Comment (coocood) [inline surf/surf_test.go]: Why use a static source?

File: surf/surf_test.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		}
+	}
+}
+
+func checkSuRF(t *testing.T, surf *SuRF, keys, vals [][]byte) {
+	for i, k := range keys {
+		val, ok := surf.Get(k)
+		require.True(t, ok)
+		require.EqualValues(t, vals[i], val)
+	}
```
Comment (coocood) [inline surf/surf_test.go]: Can we generate some non-existed keys and calculate the FPR?

File: surf/surf_test.go (hunk 1 above)
Comment (bobotu) [reply surf/surf_test.go]: Easy for reproduce bug. Or we can dump keys when test failed?

File: surf/surf_test.go (hunk 2 above)
Comment (bobotu) [reply surf/surf_test.go]: I can write a benchmark script to do this later. For measuring FPR using real world dataset such as address, email, maybe a better choice.

File: surf/surf_test.go (hunk 1 above)
Comment (coocood) [reply surf/surf_test.go]: We can use a random source and print out the source before run the test.

File: surf/surf_test.go (hunk 2 above)
Comment (coocood) [reply surf/surf_test.go]: We need to verify that the FPR is under some value.
For all the random generated key, we can pick half of the keys to build the SURF, and verify the FPR use the other half.

File: surf/surf_test.go (hunk 1 above)
Comment (bobotu) [reply surf/surf_test.go]: 👍

File: surf/surf_test.go (hunk 2 above)
Comment (bobotu) [reply surf/surf_test.go]: 👌 I’ll do this after `Iterator` merged. We need measure FPR for range queries as well.

**Comment by coocood** [comment]
LGTM

**Pull request merged**